buildpacks/nodejs-npm-install/src/npm.rs

```diff
@@ -0,0 +1,75 @@
+use commons::fun_run::CmdError;
+use libcnb::Env;
+use std::path::PathBuf;
+use std::process::Command;
+
+pub(crate) struct SetCacheConfig<'a> {
+    pub(crate) env: &'a Env,
+    pub(crate) cache_dir: PathBuf,
+}
+
+impl<'a> From<SetCacheConfig<'a>> for Command {
+    fn from(value: SetCacheConfig<'a>) -> Self {
+        let mut cmd = Command::new("npm");
+        cmd.args([
+            "config",
+            "set",
+            "cache",
+            &value.cache_dir.to_string_lossy(),
+            "--global",
+        ]);
+        cmd.envs(value.env);
+        cmd
+    }
+}
+
+#[derive(Debug)]
+pub(crate) enum VersionError {
+    Command(CmdError),
+    Parse(String, heroku_nodejs_utils::vrs::VersionError),
+}
+pub(crate) struct Version<'a> {
+    pub(crate) env: &'a Env,
+}
+
+impl<'a> From<Version<'a>> for Command {
+    fn from(value: Version<'a>) -> Self {
+        let mut cmd = Command::new("npm");
+        cmd.arg("--version");
+        cmd.envs(value.env);
+        cmd
+    }
+}
+
+pub(crate) struct Install<'a> {
+    pub(crate) env: &'a Env,
+    pub(crate) with_lockfile: bool,
+}
+
+impl<'a> From<Install<'a>> for Command {
+    fn from(value: Install<'a>) -> Self {
+        let mut cmd = Command::new("npm");
+        if value.with_lockfile {
+            cmd.arg("ci");
+        } else {
+            cmd.args(["install", "--no-package-lock"]);
```

Comment:
This was only added to maintain parity with the bash-based buildpack. I would be inclined to remove it but can we safely do that?

Reply:
I don't have data to know, just a strong hunch: most folks wouldn't notice the difference, since they are using a package-lock.json. For anyone not using a `package-lock.json`, a strong message would make sense.

Reply:
dropping this for now, will revisit in a follow up work item.